packages/charts/react-charting/src/utilities/utilities.ts

```diff
@@ -300,24 +345,55 @@ export function createStringXAxis(
 ) {
   const {
     domainNRangeValues,
-    xAxisCount = 6,
     xAxistickSize = 6,
     tickPadding = 10,
     xAxisPadding = 0.1,
     xAxisInnerPadding,
     xAxisOuterPadding,
+    containerWidth,
+    hideTickOverlap,
   } = xAxisParams;
   const xAxisScale = d3ScaleBand()
     .domain(dataset!)
     .range([domainNRangeValues.rStartValue, domainNRangeValues.rEndValue])
     .paddingInner(typeof xAxisInnerPadding !== 'undefined' ? xAxisInnerPadding : xAxisPadding)
     .paddingOuter(typeof xAxisOuterPadding !== 'undefined' ? xAxisOuterPadding : xAxisPadding);
-  const xAxis = d3AxisBottom(xAxisScale).tickSize(xAxistickSize).tickPadding(tickPadding).ticks(xAxisCount);
+
+  let tickValues = (tickParams.tickValues as string[] | undefined) ?? dataset;
+  if (hideTickOverlap) {
+    let nonOverlappingTickValues = [];
+    const tickSizes = tickValues.map(value => calculateLongestLabelWidth([value], '[class^="xAxis-"] text'));
```

Comment:
since we are computing overlap for each label here, we can relax the tickSize to actually use the value of the label and hide it if overlapping.

Reply:
Sorry for the confusion. Added a comment for clarity.